src/main/java/com/vi/counselingtoolsservice/api/service/budibase/BudibaseProxyService.java

```diff
@@ -0,0 +1,206 @@
+package com.vi.counselingtoolsservice.api.service.budibase;
+
+import com.vi.counselingtoolsservice.config.CacheManagerConfig;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotAllowedException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class BudibaseProxyService {
+
+  @Value("${budibase.proxy.host}")
+  private String proxyServiceHost;
+
+  @Value("${budibase.proxy.port}")
+  private Integer proxyServicePort;
+
+  @Value("${budibase.proxy.whitelisted}")
+  private String whitelistedURIs;
+
+  private final BudibaseApiService budibaseApiService;
+
+  public boolean isWhiteListed(HttpServletRequest request) {
+    Set<String> whitelistedURIs = new HashSet<>(Arrays.asList(this.whitelistedURIs.split(";")));
+    Optional<String> match = whitelistedURIs.stream()
+        .filter(el -> request.getRequestURI().contains(el)).findFirst();
+    return match.isPresent();
+  }
+
+  public String extractUserIdFromJWT(HttpServletRequest request) {
```

Comment:
I think this class has too many responsibilities (see single-responsibility-principle), ideally we should split it to BusibaseProxyAuthService, which contains all methods related with Roles/Authorisations and BudibaseProxyValidationService, responsible for validations

Reply:
done